app/presenters/policies_list_presenter.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class PoliciesListPresenter
+  def initialize(policies)
+    @policies = policies
+  end
+
+  def get_registry
+    return nil unless @policies
+    @policies.map do |key, value|
+      policy = value.first
+      policy.merge :name => key, :humanName => policy['name'], :data => {}
```

Comment:
I see that the purpose of this object is only this, but can't this be done in the JS execution directly so we don't have to loop it twice?

Reply:
Of course it can but it shouldn't... the less computation in the browser the better. What do you mean by _loop it twice_? Where is the first loop?

Reply:
> the less computation in the browser the better.

Good point :+1: Although the only computation for each element (the fact of being in a loop is another story that I will comment later) is the `:humanName => policy['name']`, but I trust your criteria on this.

> What do you mean by loop it twice? Where is the first loop?

I meant looping once here for this and later in the JS side in checking each element one by one to render them :smile:

Reply:
Oh but the point is precisely that: with this presenter, there will be no loop in JS. This is to prepare the data to be consumed directly by JS, which reduce the complexity in the React part significantly.

Reply:
Awesome :pray: :muscle: :1st_place_medal: 

In that case, I suggest keeping this Presenter but with the 2 improvements I mentioned above:

```ruby
# frozen_string_literal: true

class PoliciesListPresenter
  def initialize(policies)
    @policies = policies
  end

  def registry
    @registry ||= policies.map do |key, value|
      policy = value.first
      policy.merge { name: key, humanName: policy['name'], data: {} }
    end
  end

  private

   attr_reader :policies
end
```